src/java/org/apache/cassandra/db/virtual/SettingsTable.java

```diff
@@ -0,0 +1,103 @@
+package org.apache.cassandra.db.virtual;
+
+import java.lang.reflect.Field;
+import java.lang.reflect.Modifier;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Consumer;
+import java.util.stream.Collectors;
+
+import org.apache.cassandra.config.Config;
+import org.apache.cassandra.config.DatabaseDescriptor;
+import org.apache.cassandra.cql3.ColumnIdentifier;
+import org.apache.cassandra.db.DecoratedKey;
+import org.apache.cassandra.db.marshal.BooleanType;
+import org.apache.cassandra.db.marshal.UTF8Type;
+import org.apache.cassandra.db.partitions.PartitionUpdate;
+import org.apache.cassandra.db.rows.Row;
+import org.apache.cassandra.db.virtual.AbstractVirtualTable.DataSet;
+import org.apache.cassandra.dht.LocalPartitioner;
+import org.apache.cassandra.exceptions.InvalidRequestException;
+import org.apache.cassandra.schema.ColumnMetadata;
+import org.apache.cassandra.schema.TableMetadata;
+import org.apache.cassandra.service.StorageProxy;
+import org.apache.cassandra.service.StorageService;
+import org.apache.cassandra.transport.ServerError;
+
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.Lists;
+
+final class SettingsTable extends AbstractVirtualTable
+{
+    private static final String VALUE = "value";
+    private static final String SETTING = "setting";
+
+    private static final List<Field> FIELDS = Arrays.stream(Config.class.getFields())
```

Comment:
Doesn't really need to be static. Can also be a map of name -> `Field`, so that `data(DecoratedKey partitionKey)` is cleaner.